src/AWS.Deploy.Common/Recipes/Validation/RecipeValidators/ECSFargateExistingResourcesValidator.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.\r
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Threading.Tasks;
+
+namespace AWS.Deploy.Common.Recipes.Validation
+{
+    /// <summary>
+    /// Checks for existing resources that might conflict with the ECS Fargate deployment.
+    /// </summary>
+    public class ECSFargateExistingResourcesValidator : IRecipeValidator
+    {
+        public string ECSClusterName { get; set; } = "ECSCluster.NewClusterName";
+
+        /// <inheritdoc cref="ECSFargateExistingResourcesValidator"/>
+        public async Task<ValidationResult> Validate(Recommendation recommendation, IDeployToolValidationContext deployValidationContext, IOptionSettingHandler optionSettingHandler, IOptionSettingValuesFactory optionSettingValuesFactory)
+        {
+            try
+            {
+                var ecsClusterName = optionSettingHandler.GetOptionSetting(recommendation, ECSClusterName);
+                if (ecsClusterName.DisallowedValuesSource.HasValue &&
+                    optionSettingValuesFactory.GetDisallowedValues(ecsClusterName.DisallowedValuesSource.Value) is var ecsClusterNameDisallowedValues && ecsClusterNameDisallowedValues != null)
+                {
+                    var disallowedValuesList = await ecsClusterNameDisallowedValues.GetList();
+                    var ecsClusterNameValue = optionSettingHandler.GetOptionSettingValue<string>(recommendation, ecsClusterName);
+                    if (disallowedValuesList.Contains(ecsClusterNameValue))
+                        return ValidationResult.Failed($"A resource with the name '{ecsClusterNameValue}' already exists for '{ECSClusterName}'.");
+                }
+            }
```

Comment:
Wasn't this logic run via `"DisallowedValuesSource": "ECSClusterName"` on the option setting? 

If so, why repeat it in a recipe-level validator?